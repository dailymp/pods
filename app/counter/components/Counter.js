import React, { PropTypes } from 'react'

const Counter = (props) => (
  <div>
    <h2>Redux Counter:</h2>
    <p>
      <button onClick={props.onMinusClick}>-</button>
      {props.count}
      <button onClick={props.onPlusClick}>+</button>
    </p>
  </div>
)

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  onPlusClick: PropTypes.func.isRequired,
  onMinusClick: PropTypes.func.isRequired
}

export default Counter
