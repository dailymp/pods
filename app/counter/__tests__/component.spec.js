/* global describe, it, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Counter from '../components/Counter'

describe('counter component suite', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer.create(<Counter
      count={0}
      onPlusClick={jest.fn()}
      onMinusClick={jest.fn()}
    />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
