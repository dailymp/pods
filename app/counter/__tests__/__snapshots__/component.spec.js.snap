exports[`counter component suite should match the snapshot 1`] = `
<div>
  <h2>
    Redux Counter:
  </h2>
  <p>
    <button
      onClick={[Function]}>
      -
    </button>
    0
    <button
      onClick={[Function]}>
      +
    </button>
  </p>
</div>
`;
